kyo-scheduler/jvm/src/main/scala/kyo/scheduler/Scheduler.scala

```diff
@@ -0,0 +1,232 @@
+    private val flushes = new LongAdder
+
+    @volatile private var allocatedWorkers = 0
+    @volatile private var maxConcurrency   = coreWorkers
+
+    val a1, a2, a3, a4, a5, a6, a7 = 0L // padding
+
+    @volatile private var cycles = 0L
+
+    val b1, b2, b3, b4, b5, b6, b7 = 0L // padding
+
+    ensureWorkers()
+
+    private val timer = InternalTimer(timerExecutor)
+
+    private val admissionRegulator =
+        Admission(loadAvg, schedule, System.currentTimeMillis, timer)
+
+    private val concurrencyRegulator =
+        Concurrency(loadAvg, updateWorkers, Thread.sleep, System.nanoTime, timer)
+
+    def schedule(t: Task): Unit =
+        schedule(t, null)
+
+    def reject(): Boolean =
+        admissionRegulator.reject()
+
+    def reject(key: String): Boolean =
+        admissionRegulator.reject(key)
+
+    def reject(key: Int): Boolean =
+        admissionRegulator.reject(key)
+
+    def asExecutor: Executor =
+        (r: Runnable) => schedule(Task(r.run()))
+
+    def asExecutionContext: ExecutionContext =
+        ExecutionContext.fromExecutor(asExecutor)
+
+    private def schedule(t: Task, submitter: Worker): Unit =
+        @tailrec def loop(tries: Int = 0): Unit =
+            var worker: Worker = null
+            if submitter == null && tries == 0 then
+                worker = Worker.current()
+            if worker == null then
+                val m       = this.maxConcurrency
+                var i       = XSRandom.nextInt(m)
+                var stride  = Math.min(m, scheduleStride)
+                var minLoad = Int.MaxValue
+                while stride > 0 && minLoad != 0 do
+                    val w = workers(i)
+                    if w != null && !w.handleBlocking() then
+                        val l = w.load()
+                        if l < minLoad && (w ne submitter) then
+                            minLoad = l
+                            worker = w
+                    end if
+                    i += 1
+                    if i == m then
+                        i = 0
+                    stride -= 1
+                end while
+            end if
+            while worker == null do
+                worker = workers(XSRandom.nextInt(maxConcurrency))
+            if !worker.enqueue(t, force = tries >= scheduleTries) then
+                loop(tries + 1)
+        end loop
+        loop()
+    end schedule
+
+    private def steal(thief: Worker): Task =
+        var worker: Worker = null
+        var i              = 0
+        var maxLoad        = Int.MaxValue
+        while i < maxConcurrency do
+            val w = workers(i)
+            if w != null && !w.handleBlocking() then
+                val l = w.load()
+                if l > maxLoad && (w ne thief) then
+                    maxLoad = l
+                    worker = w
+            end if
+            i += 1
+        end while
+        if worker != null then
+            worker.steal(thief)
```

Comment:
Worker stealing thief not thief stealing worker?

Reply:
That's `thief` taking tasks from `worker` but yeah, the API isn't very clear.

Reply:
Maybe stealing or divisionBy